sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/FileIndexSuite.scala

```diff
@@ -178,6 +178,41 @@ class FileIndexSuite extends SharedSQLContext {
     }
   }
 
+  test("SPARK-27676: InMemoryFileIndex respects ignoreMissingFiles config for non-root paths") {
```

Comment:
There is one config `parallelPartitionDiscoveryThreshold` can control code path of partition discovery. With the default value, this only tests serial listing?

Reply:
Good point. In the case of a parallel listing, this would cause the listing Spark job to fail with a `FileNotFoundException` (after maxTaskRetries attempts to list the missing file).

In the interests of complete test coverage, I'll update the test case to exercise the parallel listing path, too. 

(Combinatorial test coverage is hard!)